### PR TITLE
Add Chrome Dev and Firefox Dev to Sauce Labs integration tests

### LINKED ISSUE #19787: Restore dev browsers in SauceLabs tests
#19782 reduced the SauceLabs batch size from 4 to 1 and disabled dev browsers. We should reactivate them and make sure tests don't become flaky again

### PR DESCRIPTION
**Pull request by estherkim**
Re-adds Chrome Dev and Firefox Dev to Sauce Labs browser mix for integration tests.

Tests are skipped by default. Can opt in with `ifChromeDev()` and `ifFirefoxDev()`.

Fixes #19787

### CODE REVIEW

File: extensions/amp-list/0.1/test/integration/test-amp-list.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ const body =
 
 const extensions = ['amp-list', 'amp-mustache', 'amp-bind'];
 
-describe('amp-list', function() {
+describe.configure().run('amp-list', function() {
```
[the remark this answers is not in the document]
Comment (estherkim) [reply extensions/amp-list/0.1/test/integration/test-amp-list.js]: Yeah, wrapping it in `run('...` is required to automatically skip the dev browsers. We could refactor out `describe.configure().run('...` of all the tests in a future PR.

File: extensions/amp-viewer-integration/0.1/test/integration/test-amp-webview-viewer-integration.js [hunk 2]
```diff
@@ -17,63 +17,64 @@
 import {AmpViewerIntegration} from '../../amp-viewer-integration';
 import {WebviewViewerForTesting} from '../webview-viewer-for-testing.js';
 
+describe.configure().run('amp-webview-viewer-integration', function() {
```
[the remark this answers is not in the document]
Comment (estherkim) [reply extensions/amp-viewer-integration/0.1/test/integration/test-amp-webview-viewer-integration.js]: See explanation in first comment.

File: extensions/amp-list/0.1/test/integration/test-amp-list.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (danielrozenberg) [reply extensions/amp-list/0.1/test/integration/test-amp-list.js]: btw, I think instead of skipping the dev browsers, we should run all tests in the equivalent dev version but make it so that failures in those browsers not fail the Travis build.

Eventually, when we get to #14835 (which I'm planning on making part of my Q1-Q2 goals) these failures will be noticed early and we can get them resolved before it's too late.

TL;DR: don't refactor the entire test suite :smile:

File: src/service/platform-impl.js [hunk 3]
```diff
@@ -16,6 +16,12 @@
 
 import {registerServiceBuilder} from '../service';
 
+// Latest stable version numbers of browsers as of 12/3/2018
+const latestVersion = {
+  chrome: 71,
+  firefox: 64,
+  safari: 12,
+};
```
[the remark this answers is not in the document]
Comment (estherkim) [reply src/service/platform-impl.js]: I'll remove all changes to `platform-impl.js`. Pretty sure you're right because this changed the bundle size and failed the check.

The link above is for Chrome Apps. For Chrome I could look for the build number, which always increases, but the implementation would be different for Firefox. Is it ok to hardcode the version numbers in `_init_tests.js` for now?

File: extensions/amp-list/0.1/test/integration/test-amp-list.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (estherkim) [reply extensions/amp-list/0.1/test/integration/test-amp-list.js]: Definitely not looking to do a big refactor :)

File: src/service/platform-impl.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rsimha) [reply src/service/platform-impl.js]: The situation I'm trying to avoid: when the next version is released, the current dev version number will become stable, and tests will suddenly stop being run on stable Chrome.

One thing to try: Debug this by printing the user agent and version name strings for Chrome, Chrome Dev, Firefox, and Firefox Dev, and see if the dev-ness of a browser version can be deduced from there.

**Comment by estherkim** [comment]
Closing in favor of #19874 that runs unstable (dev or beta) versions in warning mode.
